retworkx-core/src/traversal/dfs_visit.rs

```diff
@@ -0,0 +1,276 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+// This module is an iterative implementation of the upstream petgraph
+// ``depth_first_search`` function.
+// https://github.com/petgraph/petgraph/blob/0.6.0/src/visit/dfsvisit.rs
+
+use petgraph::visit::{
+    ControlFlow, EdgeRef, IntoEdges, Time, VisitMap, Visitable,
+};
+
+/// Return if the expression is a break value, execute the provided statement
+/// if it is a prune value.
+/// https://github.com/petgraph/petgraph/blob/0.6.0/src/visit/dfsvisit.rs#L27
+macro_rules! try_control {
```

Comment:
I think this is duplicated with: https://github.com/Qiskit/retworkx/blob/main/retworkx-core/src/traversal/bfs_visit.rs#L19

Reply:
Done in f230431.